plugins/processors/port_name/port_name.go

```diff
@@ -0,0 +1,151 @@
+package portname
+
+import (
+	"bufio"
+	"io"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/processors"
+)
+
+var sampleConfig = `
+[[processors.port_name]]
+  ## Name of tag holding the port number
+  # tag = "port"
+
+  ## Name of output tag where service name will be added
+  # dest = "service"
+
+  ## Default tcp or udp
+  # default_protocol = "tcp"
+`
+
+type sMap map[string]map[int]string // "https" == services["tcp"][443]
+
+var services sMap
+
+type PortName struct {
+	SourceTag       string `toml:"tag"`
+	DestTag         string `toml:"dest"`
+	DefaultProtocol string `toml:"default_protocol"`
+}
+
+func (d *PortName) SampleConfig() string {
+	return sampleConfig
+}
+
+func (d *PortName) Description() string {
+	return "Filter metrics with repeating field values"
+}
+
+func readServicesFile() {
+	file, err := os.Open(servicesPath())
+	if err != nil {
+		return
+	}
+	defer file.Close()
+
+	services = readServices(file)
+}
+
+func readServices(r io.Reader) sMap {
+	services = make(sMap)
+	scanner := bufio.NewScanner(r)
+	for scanner.Scan() {
+		line := scanner.Text()
+		// "http 80/tcp www www-http # World Wide Web HTTP"
+		if i := strings.Index(line, "#"); i >= 0 {
+			line = line[:i]
+		}
+		f := strings.Fields(line)
+		if len(f) < 2 {
+			continue
+		}
+		service := f[0]   // "http"
+		portProto := f[1] // "80/tcp"
+		portProtoSlice := strings.SplitN(portProto, "/", 2)
+		if len(portProtoSlice) < 2 {
+			continue
+		}
+		port, err := strconv.Atoi(portProtoSlice[0]) // "80"
+		if err != nil || port <= 0 {
+			continue
+		}
+		proto := portProtoSlice[1] // "tcp"
+
+		protoMap, ok := services[proto]
+		if !ok {
+			protoMap = make(map[int]string)
+			services[proto] = protoMap
+		}
+		protoMap[port] = service
+	}
+	return services
+}
+
+func (d *PortName) Apply(metrics ...telegraf.Metric) []telegraf.Metric {
+	for _, m := range metrics {
+		portProto, ok := m.GetTag(d.SourceTag)
+		if !ok {
+			// Nonexistent tag
+			continue
+		}
+		portProtoSlice := strings.SplitN(portProto, "/", 2)
+		l := len(portProtoSlice)
+
+		if l == 0 {
```

Comment:
I'd probably try this as a switch statement and see how it looks:
```
switch len(portProtoSlice) {
default:
case 0:
case 1:
case 2:
}
```

Reply:
I had this as a switch initially but I couldn't make it work smoothly for both cases (port only and port and protocol separated by slash) because I need to read the port in both cases.  Since golang's switch stops evaluating after the first matched case and I didn't want to copy/paste the atoi call in both cases, I switched to ifs.  I thought it didn't look much like "idiomatic go" but it works.